Sources/Client/InternetConnection.swift

```diff
@@ -0,0 +1,118 @@
+    public var onChange: OnChange = { _ in }
+    
+    /// Forces to offline mode.
+    public var offlineMode = false {
+        didSet {
+            log("✈️ Offline mode is \(offlineMode ? "On" : "Off").")
+            onChange(false)
+            
+            if offlineMode {
+                stopObserving()
+            } else if UIApplication.shared.applicationState != .background {
+                startObserving()
+            }
+        }
+    }
+    
+    /// Check if the Internet is available.
+    public var isAvailable: Bool {
+        if offlineMode {
+            return false
+        }
+        
+        let connection = reachability?.connection ?? .none
+        
+        if case .none  = connection {
+            return false
+        }
+        
+        return true
+    }
+    
+    /// Init InternetConnection.
+    init() {
+        if !isTestsEnvironment() {
+            return
+        }
+        
+        reachability?.whenReachable = { [unowned self] reachability in
+            self.log("Available 🙋‍♂️")
+            self.onChange(reachability.connection != .none)
+        }
+        
+        reachability?.whenUnreachable = { [unowned self] reachability in
+            self.log("Not Available 🤷‍♂️")
+            self.onChange(reachability.connection != .none)
+        }
+        
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(appDidBecomeActiveNotification),
+                                               name: UIApplication.didBecomeActiveNotification,
+                                               object: nil)
+        
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(appDidEnterBackgroundNotification),
+                                               name: UIApplication.didEnterBackgroundNotification,
+                                               object: nil)
+        
+        DispatchQueue.main.async { self.startObserving() }
+    }
+    
+    /// An active app state.
+    @objc private func appDidBecomeActiveNotification() {
+        if !offlineMode {
+            startObserving()
+        }
+    }
+    
+    /// A background app state.
+    @objc private func appDidEnterBackgroundNotification() {
+        stopObserving()
+    }
+    
+    // MARK: Observing Availability
+    
+    private func startObserving() {
+        do {
+            try reachability?.startNotifier()
+            log("Notifying started 🏃‍♂️")
+        } catch {
+            log("InternetConnection tried to start notifying when app state became active.\n\(error)")
+        }
+    }
+    
+    /// Stop observing the Internet connection.
+    public func stopObserving() {
+        reachability?.stopNotifier()
+        log("Notifying stopped 🚶‍♂️")
+    }
+    
+    // MARK: Logs
+    
+    private func log(_ message: String) {
+        if Client.shared.logOptions.isEnabled {
```

Comment:
Why don't we inject a `Logger` in this class?

Reply:
It doesn't have a lot of logs, no need levels. Just checks if logs enabled or not.